src/en/reference-layer-yaml.md

```diff
@@ -0,0 +1,68 @@
+Title: Charm Layer.yaml Reference  
+
+# Layer.yaml
+
+Each layer used to build a charm can have a `layer.yaml` file. The top layer
+(the one actually invoked from the command line) must. These tell the generator
+what do, ranging from which base layers to include, to which interfaces. They
+also allow for the inclusion of specialized directives for processing some
+types of files.
+
+
+### Layer Options
+
+Any layer can define options in its `layer.yaml`. Those options can then be set by other layers to change the behavior of your layer. The options are defined using [jsonschema](http://json-schema.org/), which is the same way that action
+parameters are defined.
+
+For example, the foo layer could include the following option definitions:
+
+```yaml
+includes: ['layer:basic']
+defines:  # define some options for this layer (the layer "foo")
+  enable-bar:  # define an "enable-bar" option for this layer
+    description: If true, enable support for "bar".
+    type: boolean
+    default: false
+```
+
+A layer using foo could then set it:
+
+```yaml
+includes: ['layer:foo']
+options:
+  foo:  # setting options for the "foo" layer
+    enable-bar: true  # set the "enable-bar" option to true
+```
+
+
+### Yaml Modifications
+
+config and metadata take optional lists of keys to remove from `config.yaml`
+and `metadata.yaml` when generating their data. This allows for charms to,
+for example, narrow what they expose to clients.
+
+```yaml
+Keys:
+  includes: ["layer:basic", "interface:mysql"]
+  config:
+    deletes:
+        - key names
+  metadata:
+    deletes:
+        - key names
+```
+
+
+### Custom Tactics
+
+Each file in each layer gets matched by a single Tactic. Tactics implement how
```

Comment:
WTF is a tactic?  Perhaps it would be better to lead with a brief summary of what 'tactics' are and why you should care about them - e.g. "When layers are combined in a final charm, the charm tools need to make decisions about what happens to each individual file in each layer. they do this using a set of rules known as 'tactics'...

Or whatever tactics actually are...